test/src/userobjects/InterfaceDomainUserObject.C

```diff
@@ -0,0 +1,164 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "InterfaceDomainUserObject.h"
+#include "MooseMesh.h"
+#include "Function.h"
+#include "MooseVariableFE.h"
+#include "Assembly.h"
+
+registerMooseObject("MooseTestApp", InterfaceDomainUserObject);
+
+InputParameters
+InterfaceDomainUserObject::validParams()
+{
+  InputParameters params = DomainUserObject::validParams();
+  params.addRequiredCoupledVar("u", "The variable we are diffusion for");
+  params.addRequiredCoupledVar("v", "The variable on the other side of the interface");
+  params.addParam<FunctionName>("function", "1", "The forcing function.");
+  params.addParam<Real>("robin_coef", 2, "The cofficient multiplying the Robin boundary condition");
+  params.addParam<std::vector<BoundaryName>>(
+      "robin_boundaries", "The boundaries on which to apply the Robin boundary condition");
+  params.addParam<std::vector<BoundaryName>>(
```

Comment:
We can push this parameter into the `DomainUserObject` base class. Maybe we can push above parameter in as well by renaming it as `boundaries`. Users can add a parameter for specifying boundary type. Boundary could be an interior boundary, so we may need similar treatment as interface as well.

Reply:
Maybe. In the general case, I could imagine that a user may have different operations they want to perform on different interfaces, in which case an `interface_boundaries` parameter may not be helpful at all for the user. It could be helpful for the framework code though in that it would allow us to always know whether we need to `executeOnInterface`